### PR TITLE
Show module admin tabs according to the module availability state

### LINKED ISSUE #29780: Multistore - If module is disabled in one store, its admin tabs (BO - leftmenu) is hidden everywhere else
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

Hello, We have met this problem in multistore environments. If we disable module for one of the store, his admin tabs hide in other stores. 
We made a small module for the tests:.

1. The module is enabled in both stores:
scrnli_27_09_2022_16-40-03
scrnli_27_09_2022_16-40-00

2. If I disable module for one of the store, his admin tabs hide in other store too
scrnli_27_09_2022_16-46-18
scrnli_27_09_2022_16-46-21

Our developer made a [PR](https://github.com/PrestaShop/PrestaShop/pull/29754) for this bug
Also, I attached a test module for quick tests:
[testmultistoremodule.zip](https://github.com/PrestaShop/PrestaShop/files/9657229/testmultistoremodule.zip)

### Expected behavior

If the module is disabled in one of store, his admin tabs are available in other stores

### Steps to reproduce

1. Open multistore environment
2. Install test module for both stores
3. Check modules' admin panel in both stores
4. Disable module for one of the store
5. Check modules' admin panel in both stores

### PrestaShop version(s) where the bug happened

1.7.x / 8.0.0 beta

### PHP version(s) where the bug happened

7.2 / 7.3 / 7.4

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by bogdan202**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      | In multistore env, after module deactivation in one store, his admin tabs hide in other stores as well. This PR is trying to take into account the module state while show the admin tabs
| Type?             | improvement 
| Category?         | BO
| BC breaks?        | 
| Deprecations?     | 
| Fixed ticket?     |  Fixes https://github.com/PrestaShop/PrestaShop/issues/29780
| Related PRs       | 
| How to test?      | In multistore env, deactivate the module in one store. The module tabs disappear in other stores.
| Possible impacts? |

### CODE REVIEW

**Comment by ghost** [comment]
Hello @bogdan202 

Can you open an issue for this PR, this is so that the QA team can follow the problem.

Thank you

**Comment by clotairer** [comment]
> Hello @bogdan202
> 
> Can you open an issue for this PR, this is so that the QA team can follow the problem.
> 
> Thank you

It's schedule.

**Comment by clotairer** [comment]
Associated issue  #29780 (thanks @szavadskiy)

File: src/Adapter/Module/Tab/ModuleTabManagementSubscriber.php [hunk 1]
```diff
@@ -58,8 +58,6 @@ public static function getSubscribedEvents()
         return [
             ModuleManagementEvent::INSTALL => 'onModuleInstall',
             ModuleManagementEvent::UNINSTALL => 'onModuleUninstall',
-            ModuleManagementEvent::ENABLE => 'onModuleEnable',
```
Comment (matks) [inline src/Adapter/Module/Tab/ModuleTabManagementSubscriber.php:61]: Hello @clotairer @bogdan202 and thank you for the PR and the bug report. I understand the logic you added in Tab.php but I don't understand why the changes in ModuleTabManagementSubscriber.php are needed.

I'm also afraid the changes in ModuleTabManagementSubscriber will have other consequences and might change some other behaviors 🤔

Comment (bogdan202) [reply src/Adapter/Module/Tab/ModuleTabManagementSubscriber.php:61]: Hello @matks ,
In my opinion, these modifications don't have a significant impact on other behaviors since they modify only one subscriber, and other subscribers that handle enable/disable module events keep their behavior.
This subscriber was modified because, handling the enable/disable module event, it changes  the property 'enbaled' of the admin tab for all shop contexts

Comment (matks) [reply src/Adapter/Module/Tab/ModuleTabManagementSubscriber.php:61]: Thank you for answer

**Review by matks — APPROVED** [review]

File: classes/Tab.php [hunk 2]
```diff
@@ -316,6 +316,14 @@ public static function getTabs($idLang, $idParent = null)
 
             if (is_array($result)) {
                 foreach ($result as $row) {
+                    if (!empty($row['module'])) {
+                        $module = Module::getInstanceByName($row['module']);
+                        //not to show the admin tabs of the deactivated modules
+                        if (Validate::isLoadedObject($module) && !$module->isEnabledForShopContext()) {
+                            continue;
```
Comment (atomiix) [inline classes/Tab.php:323]: What do you think about modifying the SQL query directly so we don't need to make an SQL request for each module that has a tab?

**Comment by matks** [comment]
Hi @bogdan202 

Since we had no news from you for more than 30 days, I will follow [our guidelines](https://devdocs.prestashop.com/1.7/project/maintainers-guide/reviewing-pull-requests/#stale-pull-requests) and I'll close this pull request as it seems stale.

Feel free to reopen or open another one if you think it's still relevant. You can also ping me.

Thanks!
